### ISSUE TITLE
ZA: person pages shouldn't error when the Code4SA attendance API is down

### ISSUE THREAD

**Issue by mhl** [mysociety/pombola#2023]
We've got a flood of emails from short periods when the PMG attendance API is down, and person pages on PA are down in those periods. It would be best not to to error when this is the case, and include an "attendance data temporarily unavailable, try again shortly" message.

We typically get this at line 616 of south_africa/views.py, in store_or_get_pmg_member_id
```python
    search_data = json.loads(search_resp.text)
```
... which raises a ValueError. I suggested creating an AttendanceAPIDown Exception class to raise at any point like that or others where we can catch a bad response from the API, and handling that new exception in SAPersonDetail's get_context_data.

**Comment by mhl** [comment]
Another exception we sometimes get from:

```
    search_resp = requests.get(api_search_url)
```

... is ConnectionError, which should also be turned into an AttendanceAPIDown exception

**Comment by duncanparkes** [comment]
Fixed in 243fa2a362a5b632a507e427b745b741063e3234.
